TRAIN_AND_RUN_README.md

```diff
@@ -338,6 +338,70 @@ train_command \
+two of video artifacts (e.g. for the "Locs&Confs" version, pose joint pixel coordinates 
+and confidences, the user's hand detection locations and confidences, and other procedure-relevant 
+object pixel coordinates and confidences), and outputs confidences for each of a vector 
+of activities (examples: "labels" in config/activity_labels/medical), assuming up to 
+one activity is ocurring "presently."
+
+Now, the GSP takes as input the confidence vector per frame window, and keeps track over time of which
+activities ("steps" in the GSP context) have occurred, and which step a user is on.
+
+Basically, if the "next step" at any point has been activated long enough, and with enough confidence, 
+the GSP progresses to that step as the latest "completed" step.
+
+Assumptions:
+- One activity or "background" (none of the listed activities) happens at a time.
+- The activities must happen in a specific linear order.
+- So, if an activity is detected with strong confidence way out of order (e.g. we're on step 3
+and we detect step 8), the GSP does not mark step 8 as completed.
+- A single "skipped step" is possible given some criteria. Skipping one step can be allowed unconditionally.
+Skipping one step can also be allowed if the "skipped step" has been activated with some "easier" criteria
+(a lower confidence threshold and/or fewer frames above that threshold). We can also configure to skip one step
+simply given that a threshold number of frames have passed since we completed the most recent step.
+
+## Training the GSP
+
+To "train" the GSP, we simply compute the average true positive output scores per class- that is, the 
+average confidence of each TCN activity classification in its output vector,
+only when ground truth states that activity is happening. This includes the background class.
+
+To do this, we must run inference on videos for which the TCN has never seen ground truth (and are hopefully 
+quite independent from the training videos). The validation or test splits of your dataset may suffice.
+The TCN's training harness can be run with `train=false` to only run inference and save the test data's 
+output in the needed KWCOCO output format. Example:
+
+```
+python train.py \
+    experiment=r18/feat_locsconfs \
+    paths.root_dir=/path/to/my/data/splits/ \ # See above TCN docs for training data structure
+    task_name=r18_my_TCN_i_just_trained \
+    train=false \
+    ckpt_path=model_files/activity_classifier/r18_tcn.ckpt \
+```
+
+Then, for each class, we filter the video outputs by those which the ground truth indicates that class
+activity is occurring. Then we simply average the TCN output for that activity, for those frames.
+
+This should create a new predictions output file, e.g. `tcn_activity_predictions.kwcoco.json`. 
+And you just used the ground truth for that file when you created it... That gives both files you 
+need to train the GSP, e.g. create the average TP activation numpy file:
```

Comment:
It almost seems these two paragraphs are in the opposite order? Immediately after running the `train_command` part we would have the COCO file. *Then* that filtering happens as part of running the `run_experiment` script, right?

Reply:
The wording I used there was confusing. I just added some changes to try to be more clear.